##===============================
##	 RNASeqProtocol
##===============================

#	Over the time, the methodoly becomes quite matured. Now, three main players are dominating the RNAseq quantification
#	followed by DESeq2 as the statistical moding.
#	In my effort, I have a FiveSeq strategies that ultimately provides a intersect of "significant gene" from an RAN-seq
#	Experiment.
#
#	Under this folder, I am going to focus on three quantification methods
#	kallisto, salmon, and STAR-feature count
#
#	I will have my favorite FiveSeq for extracting the significant gene list
##===========================================================================================================================


##======================================================================================================
##	I will illustrate this protocol with human NRF2 data as well as its counterpart in mouse study
##	Hopefully, with the combinatorial approcah, we can get the best and confident results
##======================================================================================================


##======================================================================================================
##      With our approach, we will bench-mark a successful RNAseq experiment, and compare the results 
##	we get to what has been published
##======================================================================================================

There are commonly accepted methods, that will be illustrated in the corresponding sub-directories:

My colleague gives great comments on this:


Pros for Salmon/Kallisto
You can give it the most comprehensive GTF/transcriptome available and it will sort out all the details
"	My favorite feature
"	You don't have to worry about overlapping genes, duplicated genes, complicated isoforms, rare isoforms, bad annotations, etc. Salmon/kallisto figure it out.
"	See below for details.

Salmon/kallisto are exceedingly good at estimating transcript abundance even for overlapping genes, and transcript isoforms with complicated exon structures.
"	Much better than featureCounts for overlapping genes
"	featureCounts cannot measure transcript expression, only gene or exon expression
"	featureCounts does not report junction read counts; Salmon/kallisto by default use junctions because they contain the best kmer specificity
"	Several publications have compared Salmon/kallisto to STAR/featureCounts (and RSEM and other methods) and Salmon/kallisto have the best ability to discover "ground truth". These pubs used simulated reads, as far as I recall.

Salmon/kallisto adjust for GC%, sequencing bias to prevent over/under-counting
"	featureCounts which reports "area under the curve", which often includes deviations based upon sequencing bias and GC% fluctuations
"	Salmon/kallisto estimates the abundance based upon multiple high-confidence points along the curve, corrected for GC% and sequencing bias
"	If you notice RNAseq coverage is "bumpy" within a gene and wondered whyÂ… these tools were designed to address the reasons (GC%, seqbias, repeats, etc.)

Salmon/kallisto appear to be very good at doing the computational work we would expect when assigning quantities to transcript isoforms
"	They assign highest quantity to isoforms that have the most supporting evidence, by using kmers specific to each isoform
"	If no evidence clearly distinguishes two isoforms, the estimate is evenly divided between them

Salmon/kallisto gene-level quantities are based upon adding isoform estimates
"	the isoform values add up proportionally to represent the sum quantity of evidence seen across the gene body, as we would expect
"	they also report "effective length" based upon actual evidence - this is not available from featureCounts

The newest versions Salmon also include "decoy" sequences
"	prevents over-counting frequently observed genomic sequences that may not be represented in the transcriptome; e.g. repeats, duplicates, rRNAs/ncRNAs not present in the index, etc.
"	this feature also fixes a previous issue of phantom counts (artifacts) that used to be assigned to some small-transcripts

A small feature is that Salmon/kallisto only use transcript FASTA file
"	no GTF necessary - if you have a novel isoform or variant, just include the FASTA sequence in the index and it works
"	it requires no upfront knowledge of overlapping transcripts or genome coordinates, it discovers by shared kmers
"	I had success using Salmon to:
o	quantify mRNA and putative enhancer RNAs together
o	quantify a mutant gene construct alongside the wildtype counterpart

You can run Salmon/kallisto with permutations to estimate confidence/variability in the quantities for each sample
"	In practice people usually do this once and never again - it's slow and limited practical value
o	partly because confidence values are already very consistent
o	partly because it is much slower
o	partly because combining within-sample variability and across-sample variability is not available in most downstream tools (only sleuth that I'm aware of)

Salmon/kallisto tend to be exceedingly fast
"	Not a true requirement, but really nice to have; each sample in 2-3 minutes, no alignment required
"	Higher accuracy, more isoform precision, much less time
"	I do usually run STAR on the side, see Cons.

Cons for Salmon/kallisto
The major "Con" is that it works only on the transcriptome you provide.
"	Note that featureCounts also only works on the GTF provided; except it can report exon-level counts.
o	Salmon/kallisto only report transcript quantities.
"	If you only "know" one isoform for a gene, it will only quantify that isoform. No novel discovery.
o	However, it will assign all expression to that isoform even if some exon-level variants exist. Gene-level values are high quality - but it cannot tell you if isoforms exist.
o	Not really different from gene-level featureCounts
o	exon-level featureCounts may be able to discover exon switching events (in my experience this doesn't work very well)
"	Best workaround is to provide the most comprehensive GTF possible, let Salmon/kallisto sort out what is actually expressed. (This was the first "Pro" point above.)
"	If you need to measure novel transcripts, you must provide those transcripts upfront.

Salmon/kallisto cannot tell you if parts of transcript coverage is "missing"
"	This is an intended feature - if coverage is missing (unmappable, high-repeat, etc.) it ignores that region. In almost all cases, this is great.
"	However, I see this with 3'UTR length:
o	if coverage in the 3'UTR is longer/shorter in some samples, Salmon/kallisto cannot tell you, *unless* you provide isoforms that also have different 3'UTR lengths.
o	Workaround is to run Salmon once as-is; re-run with two 3'UTR lengths for each isoform - this works quite well

Salmon/kallisto do not provide a convenient coverage track you can view
"	They provide "transcript-level coverage" - which would be great if there existed a transcript browser (Ha ha! Nope.)
o	you have to convert to genome coordinates; last I checked there was not an easy tool to use, I had to combine a few scripts found online
o	also transcript coverage output is optional and makes Salmon slow
"	I usually quantify expression with Salmon, and separately run STAR to make coverage tracks I can see.



I usually use "Gencode Comprehensive" GTF, and they provide a corresponding FASTA file which makes it convenient.

For example in human: https://www.gencodegenes.org/human/

The only issue is that for hg19 (GRCh37) you have to use "lift37" which they create on hg38 then lift back to hg19:
https://www.gencodegenes.org/human/release_37lift37.html




# RNASeqProtocol
Kallisto
Salmon
STAR-FeatureCounts
